docs/the-new-architecture/pillars-codegen.md

```diff
@@ -3,5 +3,227 @@ id: pillars-codegen
+
+<Tabs groupId="android-codegen" defaultValue={constants.defaultNewArchFeature} values={constants.newArchFeatures}>
+<TabItem value="turbomodules">
+
+```sh
+codegen
+├── java
+│   └── com
+│       └── MyTurbomodule
+│           └── MyTurbomodule.java
+├── jni
+│   ├── Android.mk
+│   ├── MyTurbomodule-generated.cpp
+│   ├── MyTurbomodule.h
+│   └── react
+│       └── renderer
+│           └── components
+│               └── MyTurbomodule
+│                   ├── ComponentDescriptors.h
+│                   ├── EventEmitters.cpp
+│                   ├── EventEmitters.h
+│                   ├── Props.cpp
+│                   ├── Props.h
+│                   ├── ShadowNodes.cpp
+│                   └── ShadowNodes.h
+└── schema.json
+```
+
+</TabItem>
+<TabItem value="fabric-components">
+
+```sh
+codegen
+├── java
+│   └── com
+│       └── facebook
+│           └── react
+│               └── viewmanagers
+│                   ├── MyFabricComponentManagerDelegate.java
+│                   └── MyFabricComponentManagerInterface.java
+├── jni
+│   ├── Android.mk
+│   ├── CMakeLists.txt
+│   ├── MyFabricComponent-generated.cpp
+│   ├── MyFabricComponent.h
+│   └── react
+│       └── renderer
+│           └── components
+│               └── MyFabricComponent
+│                   ├── ComponentDescriptors.h
+│                   ├── EventEmitters.cpp
+│                   ├── EventEmitters.h
+│                   ├── Props.cpp
+│                   ├── Props.h
+│                   ├── ShadowNodes.cpp
+│                   └── ShadowNodes.h
+└── schema.json
+```
+
+</TabItem>
+</Tabs>
+
+Android has some peculiarity due to the fact that `Java` can't interoperate seamlessly with `C++`. To work properly, the **CodeGen** creates some bridging within the two worlds in the `JNI` folder, where the `Java Native Interfaces` are defined.
+
+Before describing the specific files, note that both TurboModules and Fabric Components comes with a couple of build file descriptors: the `Android.mk` and the `CMakeLists.txt`. These are used by the Android app to actually build the external modules.
+
+### TurboModule
+
+The **CodeGen** generates a java abstract class in the `java` package which the same name of the TurboModule. This abstract class has to be implemented by the JNI C++ implementation.
+
+Then, it generates the `cpp` files in the `JNI` folder. They follow the same iOS convention: it creates an interface called `MyTurbomodule.h` and an implementation file called `MyTurbomodule-generated.cpp`. The former is an interface that allows React Natvie to initialize the JSI interface for the TurboModule. The latter is the implementation file which contains the logic to invoke the native method from JS and viceversa.
+
+Despite the **CodeGen** is creating a TurboModule, it still generates the files for a Fabric Component. These files are empty and they could be ignored. They will be removed soon from the Android **CodeGen**.
```

Comment:
> They will be removed soon from the Android **CodeGen**.

I would not refer to "soon" or anything like that in this document

Reply:
So I would either:

1. Remove the whole paragraph 219, as we're basically documenting a bug. It will get stale quickly once (and if) we fix it.
2. Just leave: "Despite the **CodeGen** is creating a TurboModule, it still generates the files for a Fabric Component. These files are empty and they could be ignored."

Reply:
I went for option1